homeassistant/components/binary_sensor/zha.py

```diff
@@ -87,3 +124,102 @@ def cluster_command(self, tsn, command_id, args):
+            """Initialize OnOffListener."""
+            self._entity = entity
+
+        def cluster_command(self, tsn, command_id, args):
+            """Handle commands received to this cluster."""
+            if command_id in (0x0000, 0x0040):
+                self._entity._set_state(False)
+            elif command_id in (0x0001, 0x0041, 0x0042):
+                self._entity._set_state(True)
+            elif command_id == 0x0002:
+                self._entity._set_state(not self._entity._state)
+
+        def attribute_updated(self, attrid, value):
+            """Handle attribute updates on this cluster."""
+            if attrid == 0:
+                self._entity._state = value
+                self._entity.schedule_update_ha_state()
+
+    class LevelListener:
+        """Listener for the LevelControl ZigBee cluster."""
+
+        def __init__(self, entity):
+            """Initialize LevelListener."""
+            self._entity = entity
+
+        def cluster_command(self, tsn, command_id, args):
+            """Handle commands received to this cluster."""
+            if command_id == 0x0001:  # move_to_level
+                self._entity._set_level(args[0])
+            elif command_id == 0x0002:  # step
+                # Step (technically shouldn't change on/off)
+                if args[0] == 0:
+                    self._entity._move_level(args[1])
+                else:
+                    self._entity._move_level(-args[1])
+            elif command_id == 0x0004:  # move_to_level_with_on_off
+                self._entity._set_level(args[0])
+            elif command_id == 0x0005:  # move_with_on_off
+                # We should dim slowly -- for now, just step once
+                if args[0] == 0:
+                    self._entity._move_level(10)
+                else:
+                    self._entity._move_level(-10)
+
+        def attribute_update(self, attrid, value):
+            """Handle attribute updates on this cluster."""
+            if attrid == 0:
+                self._entity._set_level(value)
+
+    def __init__(self, **kwargs):
+        """Initialize Switch."""
+        self._state = True
+        self._level = 255
+        from zigpy.zcl.clusters import general
+        self._out_listeners = {
+            general.OnOff.cluster_id: self.OnOffListener(self),
+            general.LevelControl.cluster_id: self.LevelListener(self),
+        }
+        super().__init__(**kwargs)
+
+    @property
+    def is_on(self) -> bool:
+        """Return true if the binary sensor is on."""
+        return self._state
+
+    @property
+    def device_state_attributes(self):
+        """Return the device state attributes."""
+        return {'level': self._state and self._level or 0}
+
+    def _move_level(self, change):
+        """Increment the level, setting state if appropriate."""
+        if not self._state and change > 0:
+            self._level = 0
+        self._level += min(255, max(0, self._level + change))
```

Comment:
was this meant to be `self._level = min(255, max(0, self._level + change))` ?

Reply:
Derp, yes. Thanks.